api/src/api/user.js

```diff
@@ -0,0 +1,45 @@
+const express = require('express');
+const router = express.Router();
+const { errorWrap } = require('../middleware');
+const { sendResponse } = require('../utils/response');
+const { models } = require('../models/index.js');
+const { ROLE_ENUM } = require('../utils/constants.js');
+
```

Comment:
The comments look for the post endpoint look great, but a new developer might not understand what "POST endpoint for new user" means. Can you explain more about what the endpoint tries to do without mentioning any tech jargon like "endpoint?"